RELEASE.md

```diff
@@ -20,25 +20,30 @@ Further details about versioning can be found in the [semver 2.0.0 specification
-4. `cd packages/package-name`: `cd` into the package to be released
-5. Update the version number in the `package.json`
-6. Run `yarn lint` from project root: `yarn --cwd ../../ lint`
-7. `git add package.json && git commit -m 'Release package-name@version-number'`: Commit the version bump, commit message should be `'Release package-name@version-number'` e.g. `Release web3-packagetemplate@1.0.0`
-8. `git tag package-name@version-number`: Tag the commit with package name and bumped version, e.g. `web3-packagetemplate@1.0.1`
-9. `git push origin release/package-name/bumped-version`: Push release branch to `origin`
-10. `git push --tags`: Push release tag created in `Step 8` to `origin`
-11. Create a draft release on Github similar to [this](https://github.com/ChainSafe/web3.js/releases/tag/web3-providers-base%401.0.0-alpha.1)
-    - When creating the release, select the tag created in `step 8` (`packagename@version-number`)
-    - Release title should also be `packagename@version-number`
+4. Bump packages version numbers using `lerna version --no-push --no-private --no-git-tag-version`
+    - It will prompt for new version , modify package metadata and run life cycle scripts (in our case `version`), for bootstrapping lerna will use underlying yarn.
+    - ( For first 4.x alpha version bump with lerna is not required as all versions in 4.x branch are already bumped, so invoke following scripts manually:
+      `yarn run build`)
+5. Commit the version bump changes and builds in release branch created in step 2
+6. `git tag release/bumped-version`: Tag the commit with bumped version, e.g. `4.0.0-alpha.0`
+7. `git push origin release/bumped-version`: Push release branch to `origin`
+8. `git push origin --tags`: Push release tag created in `Step 8` to `origin`
+9. Create a draft release on Github similar to [this](https://github.com/ChainSafe/web3.js/releases/tag/web3-providers-base%401.0.0-alpha.1)
+    - When creating the release, select the tag created in `step 8` (`release/bumped-version`)
+    - Release title should also be `version-number`
+    - Check `This is a pre-release`
     - In the release description, copy all entries in `CHANGELOG.md` for the version being released
-12. Click `Save draft`
-13. Open pull request to merge branch created in `Step 2` (`release/package-name/bumped-version`) into `4.x`
-14. When sufficient approvals have been met, merge the pull request
+10. Click `Save draft`
+11. Open pull request to merge branch created in `Step 2` (`release/bumped-version`) into `4.x`
+12. Wait for all tests to pass in github CI/CD actions
+13. When sufficient approvals have been met, merge the pull request
+14. Publish documentation changes
 15. Publish draft release created in `Step 11`
-16. Run `npm publish` in the package directory to publish to NPM
-    - If a tag is needed, such as `alpha`, use `npm publish --tag alpha`
+16. For each package, go in package and run `yarn publish --tag alpha` to publish to NPM
```

Comment:
Couldn't this be done using something like `lerna publish from-package --dist-tag alpha`?

Reply:
I was thinking to :
- minimum use lerna as we will remove it in future, 
- and lerna publish invokes life cycle scripts ( including version, tags push etc ) but we can avoid that using  --ignore-scripts

But its good we can use it for now as by using lerna here we dnt need to take care packages topological order during publishing.